src/shared/interfaces/categorized-validation-errors.ts

```diff
@@ -0,0 +1,6 @@
+import { SchemaValidationErrors } from './schema-validation-errors';
+
+export interface CategorizedValidationErrors {
```

Comment:
This name can be confusing, together with https://github.com/inveniosoftware-contrib/ng2-json-editor/pull/234/files#diff-20cbcdec47c6a9e70b243b42c919f425R3 , better name can be `CategorizedValidationMessages`